language-examples/remedial-js/9.0-ES5-vs-ES6/README.md

```diff
@@ -0,0 +1,961 @@
+# 9.0 - ES5 vs ES6
+
+# Purpose
+
+- JavaScript: The commonly used name for implementations of the ECMAScript standard. This term isn't tied to a particular version of the ECMAScript standard, and may be used to refer to implementations that implement all or part of any particular ECMASCript edition.
+
+- ECMAScript 5 (ES5): The 5th edition of ECMAScript, standardized in 2009. This standard has been implemented fairly completely in all modern browsers
+
+- ECMAScript 6 (ES6)/ ECMAScript 2015 (ES2015): The 6th edition of ECMAScript, standardized in 2015. This standard has been partially implemented in most modern browsers. To see the state of implementation by different browsers and tools, check out these compatibility tables:
+
+http://kangax.github.io/compat-table/es6/
+
+#### Below is a list of ES6 features along with what that feature looks like when written with the equivalent ES5 code.
+
+**Table of contents:**
+
+1. [Arrow Functions](#arrow-functions)
+1. [Block Scoping Functions](#block-scoping-functions)
+1. [Template Literals](#template-literals)
+1. [Computed Property Names](#computed-property-names)
+1. [Destructuring Assignment](#destructuring-assignment)
+1. [Default Parameters](#default-parameters)
+1. [Iterators and For-Of](#iterators-and-for-of)
+1. [Classes](#classes)
+1. [Modules](#modules)
+1. [Numeric Literals](#numeric-literals)
+1. [Property Method Assignment](#property-method-assignment)
+1. [Object Initializer Shorthand](#object-initializer-shorthand)
+1. [Rest Parameters](#rest-parameters)
+1. [Spread Operator](#spread-operator)
+1. [Proxying a function object](#proxying-a-function-object)
+1. [Array-like object to array](#array-like-object-to-array)
+1. [About](#about)
+1. [License](#license)
+
+## Arrow Functions
+
+An arrow function expression (also known as fat arrow function) has a shorter syntax compared to function expressions and lexically binds the this value. Arrow functions are always anonymous.
+
+
+ES6:
+
+```js
+[1, 2, 3].map(n => n * 2);
+// -> [ 2, 4, 6 ]
+```
+
+ES5 equivalent:
+
+```js
+[1, 2, 3].map(function(n) { return n * 2; }, this);
```

Comment:
All that we're saying here is that the this value is based solely on the surrounding scope, but I'll actually add that explanation verbatim.